### ISSUE TITLE
Smartmatch always fails on two Seq objects.

### ISSUE THREAD

**Issue by jmaslak** [rakudo/rakudo#2233]
## The Problem

This is similar to #2218 

Two Sequence objects won't smartmatch.  Example, using the output of sort (which produces a Seq):

```
 my @list = (1, 2, 3);
 my $seq1 = @list.sort;  # produce a Seq object
 my $seq2 = @list.sort;  # produce a Seq object
 say $seq1 ~~ $seq2;     # outputs False, not as expected
```

## Expected Behavior

I think a Seq should be able to smartmatch another Seq.

## Actual Behavior

Seq will not smartmatch Seq's with identical content.

## Steps to Reproduce

See above.

## Environment
* Operating system: Ubuntu 18.04
* Compiler version (`perl6 -v`): moar-master via rakudobrew on current master.

```
This is Rakudo version 2018.06-462-g5ee9afaea built on MoarVM version 2018.06-428-g727767006
implementing Perl 6.c.
```

**Comment by lizmat** [comment]
FWIW, that would mean that both Seq’s would need to be produced / cached.  Is that what you want?

**Comment by jnthn** [comment]
@lizmat I agree that the `Seq` would need to be consumed, but I don't think it needs to be cached; can't we proceed pairwise iterating each side, and return `False` if the pairs don't smart match or if they are of unequal length? Possibly if one or more of the inputs is lazy we'd want to give an error.

**Comment by jnthn** [comment]
(To be clear, I'm proposing a general semantic for `Iterable` `~~` `Seq` here, which would cover `Seq` `~~` `Seq`.)

**Comment by lizmat** [comment]
Ok, if you think that's a valid approach, I'll make that happen (soonish)
